packages/mermaid/src/mermaidAPI.ts

```diff
@@ -285,7 +285,9 @@ export const cleanUpSvgCode = (
  * TODO replace btoa(). Replace with  buf.toString('base64')?
  */
 export const putIntoIFrame = (svgCode = '', svgElement?: D3Element): string => {
-  const height = svgElement ? svgElement.viewBox.baseVal.height + 'px' : IFRAME_HEIGHT;
+  const height = svgElement?.viewBox?.baseVal?.height
+    ? svgElement.viewBox.baseVal.height + 'px'
+    : IFRAME_HEIGHT;
```

Comment:
I've changed [`/demos/info.html`](https://github.com/mermaid-js/mermaid/blob/3d117815425cbd05105ec0f0192adc0be91931f2/demos/index.html) config to this:

```diff
<script type="module">
  import mermaid from './mermaid.esm.mjs';
  mermaid.initialize({
    theme: 'forest',
    logLevel: 3,
-   securityLevel: 'loose',
+   securityLevel: 'sandbox',
  });
</script>
```

The first would fail, but others would work fine:

![image](https://github.com/mermaid-js/mermaid/assets/87622592/b2b003ad-72f7-4dbf-ba58-bf6b85cc9d1c)

Not sure why yet, maybe a variable has been defined after first one, that's why `e2e` testing is failing.

Reply:
The above screenshot is in firefox, but it works fine in chrome:

![image](https://github.com/mermaid-js/mermaid/assets/87622592/fb4c58ea-714e-491b-8eb0-96ad4726ebf0)

Reply:
After more debugging, when I added the following line of code in [`/packages/diagrams/info/infoRenderer.ts`](https://github.com/mermaid-js/mermaid/blob/3d117815425cbd05105ec0f0192adc0be91931f2/packages/mermaid/src/diagrams/info/infoRenderer.ts#L21-L24):

```diff
if (securityLevel === 'sandbox') {
  const sandboxElement: HTML = select('#i' + id);
  document = sandboxElement.node()?.contentDocument;
+  log.info(document?.body)
}
```

It shows that the body is being use in the first time is the parent one, not sub body of `iframe` element:

![image](https://github.com/mermaid-js/mermaid/assets/87622592/c2d95cad-74a0-444e-8c99-c14c12e51b1e)

Reply:
What do you think the cause is?

Reply:
I don't know. The weird part that it works fine in chrome.

Reply:
@sidharthv96 Let's leave the `sandbox` part for later and create an issue with the above details, then figure out why it happened and how to resolve it so we can merge this PR.